src/overlays/actors/ovl_En_Mag/z_en_mag.c

```diff
@@ -27,21 +33,553 @@ const ActorInit En_Mag_InitVars = {
+        gDPLoadTextureBlock(gfx++, D_06019A00, G_IM_FMT_IA, G_IM_SIZ_8b, 160, 16, 0, G_TX_NOMIRROR | G_TX_CLAMP,
+                            G_TX_NOMIRROR | G_TX_CLAMP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+
+        gSPTextureRectangle(gfx++, 312, 792, 952, 856, G_TX_RENDERTILE, 0, 0, 1024, 1024);
+    }
+
+    if (gSaveContext.fileNum == 0xFEDC) {
+        // Draw "NO CONTROLLER" Text
+        textAlpha = textFadeTimer * 10;
+        if (textAlpha >= 255) {
+            textAlpha = 255;
+        }
+
+        // Text Shadow
+        gDPPipeSync(gfx++);
+        gDPSetCombineLERP(gfx++, 0, 0, 0, PRIMITIVE, TEXEL0, 0, PRIMITIVE, 0, 0, 0, 0, PRIMITIVE, TEXEL0, 0, PRIMITIVE,
+                          0);
+        gDPSetPrimColor(gfx++, 0, 0, 0, 0, 0, textAlpha);
+
+        rectLeft = VREG(19) + 1;
+        for (i = 0; i < 12; i++) {
+            EnMag_DrawTextFont(&gfx, buf + 0x3C88 + (noControllerFontIndexes[i] * 0x80), rectLeft, YREG(10) + 172);
+            rectLeft += VREG(21);
+            if (i == 1) {
+                rectLeft += VREG(23);
+            }
+        }
+
+        // Actual Text
+        gDPPipeSync(gfx++);
+        gDPSetPrimColor(gfx++, 0, 0, 100, 255, 255, textAlpha);
+
+        rectLeft = VREG(19);
+        for (i = 0; i < 12; i++) {
+            EnMag_DrawTextFont(&gfx, buf + 0x3C88 + (noControllerFontIndexes[i] * 0x80), rectLeft, YREG(10) + 171);
+            rectLeft += VREG(21);
+            if (i == 1) {
+                rectLeft += VREG(23);
+            }
+        }
+    } else if (this->copyrightAlpha >= 200.0f) {
+        // Draw "PRESS START" Text
+        textAlpha = textFadeTimer * 10;
+        if (textAlpha >= 255) {
+            textAlpha = 255;
+        }
+
+        // Text Shadow
+        gDPPipeSync(gfx++);
+        gDPSetCombineLERP(gfx++, 0, 0, 0, PRIMITIVE, TEXEL0, 0, PRIMITIVE, 0, 0, 0, 0, PRIMITIVE, TEXEL0, 0, PRIMITIVE,
+                          0);
+        gDPSetPrimColor(gfx++, 0, 0, 0, 0, 0, textAlpha);
+
+        rectLeft = YREG(7) + 1;
+        for (i = 0; i < 10; i++) {
+            EnMag_DrawTextFont(&gfx, buf + 0x3C88 + (pressStartFontIndexes[i] * 0x80), rectLeft, YREG(10) + 172);
+            rectLeft += YREG(8);
+            if (i == 4) {
+                rectLeft += YREG(9);
+            }
+        }
+
+        // Actual Text
+        gDPPipeSync(gfx++);
+        gDPSetPrimColor(gfx++, 0, 0, YREG(4), YREG(5), YREG(6), textAlpha);
+
+        rectLeft = YREG(7);
+        for (i = 0; i < 10; i++) {
+            EnMag_DrawTextFont(&gfx, buf + 0x3C88 + (pressStartFontIndexes[i] * 0x80), rectLeft, YREG(10) + 171);
+            rectLeft += YREG(8);
+            if (i == 4) {
+                rectLeft += YREG(9);
+            }
+        }
+    }
+
+    if (textFadeDirection != 0) {
+        if (--textFadeTimer == 0) {
+            textFadeDirection = 0;
+        }
+    } else {
+        if (++textFadeTimer >= 26) {
+            textFadeDirection = 1;
+        }
+    }
```

Comment:
I considered making it a boolean at first, but since it's used to indicate whether the fade timer is increasing or decreasing (and by extension the text alpha), I think it makes more sense as a direction.